packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth/methods/call.ts

```diff
@@ -695,6 +698,85 @@ contract C {
           }
         });
 
+        describe("sender balance", function () {
+          const sender = "0x1234512345123451234512345123451234512345";
+          let contractAddress: string;
+
+          beforeEach("fund the sender", async function () {
+            contractAddress = await deployContract(
+              this.provider,
+              `0x${EXAMPLE_READ_CONTRACT.bytecode.object}`
+            );
+          });
+
+          it("should use the sender's balance if it's enough", async function () {
+            // fund the sender
+            await this.provider.send("eth_sendTransaction", [
+              {
+                from: DEFAULT_ACCOUNTS_ADDRESSES[0],
+                to: sender,
+                value: numberToRpcQuantity(1_000_000),
+              },
+            ]);
+            await assertAddressBalance(this.provider, sender, 1_000_000n);
+
+            // call the contract
+            const senderBalance = await this.provider.send("eth_call", [
+              {
+                from: sender,
+                to: contractAddress,
+                data: EXAMPLE_READ_CONTRACT.selectors.senderBalance,
+                gas: numberToRpcQuantity(100_000),
+                gasPrice: numberToRpcQuantity(1),
+              },
+            ]);
+
+            assert.equal(rpcDataToNumber(senderBalance), 900_000);
+          });
+
+          it("should increase the sender's balance if it's positive but not enough", async function () {
```

Comment:
Should it really? I'm not convinced that this is the expected behavior.

Reply:
but _is it_ increasing the sender's balance? It's funded to `1_000`, then checks at the end that the balance is `0`, meaning that it actually _decreased_ when the sender's balance was positive but not enough. maybe this a typo/mistake from an earlier draft or something?

Reply:
Btw, if you both are fine with this, then I will add some explanatory comments to these tests.

If you think the behavior should be different... then that will require changes in ethereumjs.

Reply:
Comments would be much appreciated, thanks. (I'll have to defer to others on whether the behavior should be different.)

Reply:
Done.